applications/serial_lte_modem/src/gnss/slm_at_gnss.c

```diff
@@ -385,16 +385,26 @@ static void cell_pos_req_wk(struct k_work *work)
 
 	ARG_UNUSED(work);
 
-	if (cell_pos_type == CELL_POS_TYPE_SINGLE) {
-		err = nrf_cloud_cell_pos_request(NULL, true, NULL);
+	if (cell_pos_type == LOCATION_TYPE_SINGLE_CELL) {
+		struct lte_lc_cells_info scell_inf = {.ncells_count = 0};
+
+		/* Obtain the single cell info from the modem */
+		err = nrf_cloud_location_scell_data_get(&scell_inf.current_cell);
+		if (err) {
+			LOG_ERR("Failed to obtain cellular network info, error: %d", err);
+			return;
+		}
+
+		/* Request location using only the single cell info */
+		err = nrf_cloud_location_request(&scell_inf, NULL, true, NULL);
 		if (err) {
 			LOG_ERR("Failed to request SCELL, error: %d", err);
 		} else {
 			LOG_INF("nRF Cloud SCELL requested");
 		}
 	} else {
```

Comment:
currently SLM only supports cellular positioning.
it explicitly sets `cell_pos_type` to either SCELL or MCELL.
so i feel it is fine as is, but i will defer to @junqingzou since he is the SLM owner.

Reply:
> it explicitly sets cell_pos_type to either SCELL or MCELL.

I did see that -- the suggestion is more aimed at improving readability -- it isn't obvious to someone reading that section of code that the other branch is guaranteed to execute only with MCELL, unless that person happens to also have read the bit of code that actually sets SCELL/MCELL. Whereas if you make the cases explicit, a reader doesn't have to do that to understand the cases

But you're right, it's really up to @junqingzou

Reply:
yes, making this `else if (cell_pos_type == LOCATION_TYPE_MULTI_CELL) {` would be good.